ssl/ssl_ciph.c

```diff
@@ -493,8 +493,47 @@ int ssl_cipher_get_evp_md_mac(SSL_CTX *ctx, const SSL_CIPHER *sslc,
     return 1;
 }
 
+int ssl_cipher_get_evp_cipher_ecb(SSL_CTX *ctx, const SSL_CIPHER *sslc,
+                                  const EVP_CIPHER **enc)
+{
+    int i = ssl_cipher_info_lookup(ssl_cipher_table_cipher, sslc->algorithm_enc);
+
+    if (i == -1) {
+        *enc = NULL;
+    } else {
+        if (i == SSL_ENC_NULL_IDX) {
+            /*
+             * We assume we don't care about this coming from an ENGINE so
+             * just do a normal EVP_CIPHER_fetch instead of
+             * ssl_evp_cipher_fetch()
+             */
+            *enc = EVP_CIPHER_fetch(ctx->libctx, "NULL", ctx->propq);
+        } else {
+            char *ecb_name = NULL;
+
+            if ((sslc->algorithm_enc & SSL_AES128_ANY) != 0) {
+                ecb_name = "AES-128-ECB";
+            } else if ((sslc->algorithm_enc & SSL_AES256_ANY) != 0) {
+                ecb_name = "AES-256-ECB";
+            } else if (ossl_assert((sslc->algorithm_enc & SSL_CHACHA20) != 0)) {
+                ecb_name = "ChaCha20";
+            }
```

Comment:
Let's perhaps capture and save the related data (iv length and input length) here and not do ad hoc inspection of the cipher name later in `dtls_crypt_sequence_number()`.

A more general abstraction than saving away iv and input lengths, ... is  a function pointer for doing the sequence number encryption for each supported family.

The relevant lengths (or more general function pointer) could be statically defined as part of each TLS cipher (with those that don't provide usable values then not suitable for use with DTLS 1.3). The code here would then be algorithm independent.

Reply:
Now I capture one additional variable namely the offset. And from that I can determine the other parameters.